### ISSUE TITLE
Queue_free()'ing a BoneAttachment3D in an exported build throws a signal error

### ISSUE THREAD

**Issue by Michaelcraighoffman** [godotengine/godot#87804]
### Tested versions

Tested in:
v4.1.1.stable.mono.official [bd6af8e0e]
v4.1.4.rc1.mono.official [b9008f3d5]
v4.2.1.stable.mono.official [b09f793f5]
v4.2.2.rc1.mono.official [c7fb0645a]

### System information

Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3080 (NVIDIA; 31.0.15.3598) - 12th Gen Intel(R) Core(TM) i5-12600K (16 Threads)

### Issue description

In version 4.1.1 and 4.1.4, no error appears. 

In version 4.2.1 and 4.2.2:
When a project is exported and a node containing a skeleton3D with a BoneAttachedment3D is deleted from the tree with queue_free(), the following error consistently appears:

```
ERROR: Attempt to disconnect a nonexistent connection from 'Skeleton3D:<Skeleton3D#23722984641>'. Signal: 'bone_pose_changed', callable: ''. 
at: (core/object/object.cpp:1420)
```

### Steps to reproduce

1. Add a glb model containing a skeleton
2. Create an inherited scene from the imported model
3. Add a BoneAttachment3D node to the skeleton in that inherited scene
4. Instantiate the scene as a node
5. Queue_free() the node

### Minimal reproduction project (MRP)

[repro.zip](https://github.com/godotengine/godot/files/14119017/repro.zip)

**Comment by bitsawer** [comment]
I don't think this is a Godot bug, at least not the issue in the MRP project. The test project gives an error in Godot debug builds too because you are trying to `queue_free()` the same node multiple times. The line in documentation about `"It is also safe to call queue_free multiple times."` means it's safe to call it multiple times before the end of the frame. Add line `character = null` to prevent calling the method on a freed object. So the new function would look like this:

```gdscript
# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta):
    remaining-=delta
    if(remaining < 0 && character!=null):
        character.queue_free()
        character = null # <- Add this line.
        print("Removed")
```

**Comment by Michaelcraighoffman** [comment]
The test project does not display any errors on any version for me in the editor and I've verified multiple times that both my original project and the MRP one do not call queue_free() multiple times.  I have nonetheless made the change you requested.  The behavior on all 6 versions of godot remains unchanged. 4.1.1 and 4.1.4 show no errors in the editor or in exported builds.  4.2, 4.2.1, 4.2.2, and 4.3dev2 all work fine in the editor, but display the error in exported builds.

**Comment by bitsawer** [comment]
I mentioned I used Godot debug build build using current master (commit at f8a039e9b54875a3435ace4b9953cefa591a0753). Normal and release builds have different level of error checking and reporting, especially when handling freed objects. The test project seems to work (or at least not crash) in official builds (at least 4.2) without errors, but gives an error when using a debug build in current master branch:

`Attempt to call function 'queue_free' in base 'previously freed' on a null instance.`

I haven't tested an export build, so there might still be a bug somewhere in Godot.

![capture](https://github.com/godotengine/godot/assets/22456603/ff83d192-f506-4397-8ebc-9fc2450f7267)

**Comment by Michaelcraighoffman** [comment]
A quick follow-up clarification, this error has not led to any crashes for me and the nodes do seem to be correctly removed from the tree

**Comment by akien-mga** [comment]
@Michaelcraighoffman Do you see the error from @bitsawer's screenshot when running the project in the editor?

If so, I think this is not a bug and working as intended. As mentioned, release and debug builds have different levels of error checking for performance reasons, and the error in the debug build (editor) is the most pertinent one, and something that *must* be fixed to avoid getting errors (or even crashes) in release exports.

**Comment by Michaelcraighoffman** [comment]
@akien-mga After resolving the error bitsawer mentioned, I still see the original reported error regarding BoneAttachments only in exported builds.  I have not had time to test this issue in newer Godot Builds.
